Config/CMakeIncludes/sanity_check_and_xsdk.cmake

```diff
@@ -174,20 +177,42 @@ if (Tasmanian_ENABLE_MPI)
     endif()
 endif()
 
-# check for fortran, note that enable_language always gives FATAL_ERROR if compiler is missing (no way to recover or autodisable)
+# check for Fortran, note that enable_language always gives FATAL_ERROR if the compiler is missing (no way to recover or autodisable)
 if (Tasmanian_ENABLE_FORTRAN)
     enable_language(Fortran)
 endif()
 
-# debug flags and workarounds (Fortran doesn't know how to handle those)
-if (${CMAKE_BUILD_TYPE} STREQUAL "Debug")
-    add_definitions(-D_TASMANIAN_DEBUG_)
-    if (Tasmanian_ENABLE_FORTRAN) # AND (${CMAKE_Fortran_COMPILER_ID} STREQUAL "GNU"))
-        set(CMAKE_Fortran_FLAGS_DEBUG "-g")
-    endif()
-else()
-    # this is a bug in some versions of CUDA
-    if ((Tasmanian_ENABLE_CUBLAS OR Tasmanian_ENABLE_CUDA) AND ("${CUDA_VERSION_MAJOR}" LESS 8))
-        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -D_FORCE_INLINES")
+
+########################################################################
+# Compiler specific flags: Intel hasn't been tested in a while
+# Tasmanian_STRICT_OPTIONS=ON will prevent Tasmanian from setting
+# compiler flags, only the user provided flags will be used
+########################################################################
+if (NOT Tasmanian_STRICT_OPTIONS)
+    if ((${CMAKE_CXX_COMPILER_ID} STREQUAL "GNU") OR (${CMAKE_CXX_COMPILER_ID} STREQUAL "Clang"))
+        set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -O3") # there is no point in making a slow debug
+        if (Tasmanian_ENABLE_FORTRAN)
+            set(CMAKE_Fortran_FLAGS_DEBUG "${CMAKE_Fortran_FLAGS} -O3 -fno-f2c")
+        endif()
+#    elseif (${CMAKE_CXX_COMPILER_ID} STREQUAL "Intel")
+#        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -O3 -mtune=native -diag-disable 11074 -diag-disable 11076 -Wall -Wextra -Wshadow -Wno-unused-parameter -pedantic")
+    elseif (${CMAKE_SYSTEM_NAME} STREQUAL "Windows") # TODO: VC++ seems to ignore the cmake CXX flags
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /Ox /EHsc")
     endif()
 endif()
+
+
+########################################################################
+# Extra directories:
+# sometimes find_package() fails to recognize all dependencies
+# the extra variables here allow the user to circumvent the problem by
+# including additional directories, also check SparseGrids/CMakeLists.txt
+# comment about Tasmanian_EXTRA_LIBRARIES
+########################################################################
+if (DEFINED Tasmanian_EXTRA_INCLUDE_DIRS)
+    include_directories(${Tasmanian_EXTRA_INCLUDE_DIRS})
+endif()
+
+if (DEFINED Tasmanian_EXTRA_LINK_DIRS)
+    link_directories(${Tasmanian_EXTRA_LINK_DIRS})
```

Comment:
Same comment as for `include_directories` here.  Why did you choose to do this over `target_link_directories()`?

Reply:
I should have done that. I will do it.

Reply:
I will use `target_include_directories()`, but `link_directories()` cannot be specified per-target. Those should rarely be necessary, but can give you a fighting chance when dealing with bugs or exotic setup.